docs/sources/static/configuration/agent-management.md

```diff
@@ -0,0 +1,123 @@
+
+  # Protocol to use when connecting to the API server (http|https).
+  protocol: <string>
+
+  # The polling interval for fetching the configuration.
+  polling_interval: <string>
+
+  # Sets the `Authorization` header on every request with the
+  # configured username and password.
+  basic_auth:
+    [ username: <string> ]
+    [ password_file: <string> ]
+
+  # Fields specific to remote configuration.
+  remote_configuration:
+    # A path to a directory where the remote configuration will be cached. The directory must be writeable.
+    cache_location: <string>
+
+    # The namespace to use.
+    namespace: <string>
+
+    # Set of self-identifying labels used for snippet selection.
+    labels:
+      [ <labelname>: <labelvalue> ... ]
+```
+
+## API (v2)
+
+Grafana Agents with Agent Management enabled continuously poll the API server for an up-to-date configuration. The API server is expected to return an HTTP response with the following body format:
+
+```yaml
+# The base configuration for the Agent.
+base_config: |
+  <grafana_agent_config>
+# A set of snippets to be conditionally merged into the base configuration.
+snippets:
+  [ <snippet_name>: <snippet_content> ... ]
+```
+
+### grafana_agent_config
+
+This is a standard Grafana Agent [static mode configuration](https://grafana.com/docs/agent/latest/static/configuration/). Typically used to configure the server, remote_writes, and other global configuration.
+
+### snippet_content
+
+The snippet content is a YAML document which conforms to the following schema:
+
+```yaml
+# Config provides the actual snippet configuration.
+config: |
+  [metrics_scrape_configs]:
+  - [<scrape_config> ... ]
+  [logs_scrape_configs]:
+  - [<promtail.scrape_config> ... ]
+  [integration_configs]:
+    [<integrations_config> ... ]
+# Selector is a set of labels used to decide which snippets to apply to the final configuration.
+selector:
+  [ <labelname>: <labelvalue> ... ]
+```
+
+> **Note:** More information on the following types can be found in their respective documentation pages:
+>
+> * [`scrape_config`](https://prometheus.io/docs/prometheus/2.37/configuration/configuration/#scrape_config)
+> * [`promtail.scrape_config`](https://grafana.com/docs/loki/latest/clients/promtail/configuration/#scrape_configs)
+> * [`integrations_config`](https://grafana.com/docs/agent/latest/static/configuration/integrations)
+
+
+### Example response body
+
+```yaml
+base_config: |
+  server:
+    log_level: debug
+snippets:
+  snip1:
+    config: |
+      metrics_scrape_configs:
+      - job_name: 'prometheus'
+        scrape_interval: 15s
+        static_configs:
+        - targets: ['localhost:9090']
+      logs_scrape_configs:
+      - job_name: 'loki'
+        static_configs:
+        - targets: ['localhost:3100']
+      integration_configs:
+        node_exporter:
+          enabled: true
+    selector:
+      os: linux
+      app: app1
+```
+
+Note: both json and yaml responses are supported.
```

Comment:
```suggestion
Note: both JSON and YAML responses are supported.
```

What determines getting a JSON/YAML response? Accept/Content-Type header?

Reply:
I will add a note, but yes, via `Accept` header, defaulting to JSON